assets/src/design-system/components/menu/menu.js

```diff
@@ -47,22 +49,27 @@ import { EmptyList, ListGroupings } from './list';
  *
  */
 
-const Menu = ({
-  dropDownHeight,
-  emptyText,
-  menuStylesOverride,
-  hasMenuRole,
-  isRTL,
-  options = [],
-  listId,
-  onMenuItemClick,
-  onDismissMenu,
-  renderItem,
-  activeValue,
-  menuAriaLabel,
-  parentId,
-}) => {
-  const listRef = useRef();
+const Menu = (
+  {
+    dropDownHeight,
+    emptyText,
+    menuStylesOverride,
+    hasMenuRole,
+    handleReturnToParent,
+    isMenuFocused = true,
+    isRTL,
+    options = [],
+    listId,
+    onMenuItemClick,
+    onDismissMenu,
+    renderItem,
+    activeValue,
+    menuAriaLabel,
+    parentId,
+  },
+  ref
+) => {
+  const listRef = ref;
```

Comment:
💭 After playing with this and looking at the stories I think we should define the `listRef` inside this menu component rather than passing in a ref from the outside.

It was smelly to me because we were defining the `menuRef` in the story file without doing anything with it. If we need to do this for every usage of the menu, we should move it down into the component so that it doesn't have to be defined each time we use it.
<image src="https://user-images.githubusercontent.com/22185279/107276778-c1e58e80-6a10-11eb-9283-9c19c2460bbd.png" width="450px" />

Reply:
```suggestion
  const listRef = useRef(null);
```

Reply:
Thanks for sticking up for this, the reason it was getting passed in was because i was using it in `Search` to send to the menu and if it wasn't getting forwarded it was attaching to out of date DOM, but this is way cleaner now and this was absolutely smelly. Fixing :)